components/schema-migrator/migrations/director/20231211174610_add_aspect_event_resource.up.sql

```diff
@@ -0,0 +1,125 @@
+BEGIN;
+
+-- Drop views --
+DROP VIEW IF EXISTS aspect_event_resources_subset;
+DROP VIEW IF EXISTS aspect_event_resources;
+DROP VIEW IF EXISTS aspects_tenants; -- this view will not be changed but it depends on the column aspects.event_resources; that's why the view has to be dropped and re-created in order to drop the column
+DROP VIEW IF EXISTS tenants_aspects;
+
+-- Create aspect_event_resources table
+CREATE TABLE aspect_event_resources (
+                         id UUID PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+                         aspect_id UUID NOT NULL,
+                         CONSTRAINT aspect_event_resources_aspect_id_fkey FOREIGN KEY (aspect_id) REFERENCES aspects (id) ON DELETE CASCADE,
+                         app_id UUID,
+                         CONSTRAINT aspect_event_resources_app_id_fkey FOREIGN KEY (app_id) REFERENCES applications (id) ON DELETE CASCADE,
+                         app_template_version_id UUID,
+                         CONSTRAINT aspect_event_resources_app_template_version_id_fk FOREIGN KEY (app_template_version_id) REFERENCES app_template_versions (id) ON DELETE CASCADE,
+                         ord_id VARCHAR(256) NOT NULL,
+                         CONSTRAINT aspect_event_resources_ord_id_unique UNIQUE (app_id, ord_id),
```

Comment:
We can add the aspect_id in the constraint tuple so that the uniqueness can be in the scope of the aspect. If it is only (app_id, ord_id) when we want to create integration dependencies with at least one similar aspect this constraint will be hit.